app/main/api_caller.py

```diff
@@ -0,0 +1,38 @@
+import os
+import datetime as dt
+import requests
+
+MOVIE_DB_IMAGE_URL = "https://image.tmdb.org/t/p/w500"
+MOVIE_API_URL = "https://api.themoviedb.org/3"
+
+
```

Comment:
I think basically it is ok but I see code repetition here and in next step we are going to add retries to requests in case of 500

what do you think about extracting making requests:

```suggestion
    def make_request(self, url, params=None)-> dict:
        response = requests.get(url, headers=self.headers, params=params)
        response.raise_for_status()  # Raises an HTTPError for bad responses
        return response.json()

    def get_movie(self, movie_id):
        url = os.path.join(MOVIE_API_URL, "movie", f"{movie_id}?language=en-US")
        response_data = self.make_request(url)
        release_date = response["release_date"]
        parsed_date = dt.datetime.strptime(release_date, "%Y-%m-%d")
        year = parsed_date.year

        new_movie_data = {
            "title": response["original_title"],
            "year": year,
            "description": response["overview"],
            "img_url": f"{MOVIE_DB_IMAGE_URL}{response['poster_path']}",
        }
        return new_movie_data

    def get_movies_to_select(self, title):
        url = os.path.join(MOVIE_API_URL, "search", "movie")
        params = {"query": f"{title}"}
        response_data = self.make_request(url, params=params)
        return response_data.get("results", [])
```

Reply:
I think that's great idea! It'll make code easier to read as well.